### PR TITLE
Refactor mnist/train_test.py from testing single epoch to single train step

### PR DESCRIPTION
**Pull request by mohitreddy1996**
This change is to make the tests across examples consistent.

- GNNs: https://github.com/google/flax/pull/224
- Issue: https://github.com/google/flax/issues/176

### CODE REVIEW

File: examples/mnist/train_test.py [hunk 1]
```diff
@@ -21,26 +21,23 @@
...
+  def test_single_train_step(self):
+    train_ds, _ = train.get_datasets()
+    batch_size = 32
     model = train.create_model(random.PRNGKey(0))
     optimizer = train.create_optimizer(model, 0.1, 0.9)
-    optimizer, train_metrics = train.train_epoch(optimizer, train_ds, 128, 0,
-                                                 input_rng)
-    self.assertLessEqual(train_metrics['loss'], 0.27)
-    self.assertGreaterEqual(train_metrics['accuracy'], 0.92)
-    loss, accuracy = train.eval_model(optimizer.target, test_ds)
```
Comment (avital) [inline examples/mnist/train_test.py:41]: I think this test should still execute the `eval_model` step even thought we're only training for one step.

File: examples/mnist/train_test.py [hunk 2]
```diff
@@ -21,26 +21,23 @@
...
-    self.assertLessEqual(loss, 0.06)
-    self.assertGreaterEqual(accuracy, 0.98)
+
+    _, train_metrics = \
```
Comment (avital) [inline examples/mnist/train_test.py:36]: I know this should be fixed with a linter and not a human :) but I don't think we use backslashes anywhere else in our code like this for newline. I would do

```
_, train_metrics = train.train_step(optimizer,
                                    batch=...)
```

File: examples/mnist/train_test.py (hunk 1 above)
Comment (avital) [reply examples/mnist/train_test.py:41]: (For MNIST it should be fine. For larger datasets that would be too slow, so we should at least test `eval_model` on a subset of the dataset -- say the first 32 elements)

Comment (mohitreddy1996) [reply examples/mnist/train_test.py:41]: Yup that sounds good. Done!

File: examples/mnist/train_test.py (hunk 2 above)
Comment (mohitreddy1996) [reply examples/mnist/train_test.py]: Done, my bad. Made the Indentation consistent with train.py

**Pull request merged**